src/goto-programs/goto_trace.h

```diff
@@ -138,6 +136,22 @@ class goto_trace_stept
     full_lhs_value.make_nil();
     cond_expr.make_nil();
   }
+
+  goto_programt::const_targett get_pc() const { return pc; }
+
+  /// We don't have a way of enforcing this, but \p pc_ must *not* be a
+  /// singular iterator!
+  void set_pc(goto_programt::const_targett pc_)
+  {
+    pc = std::move(pc_);
+    is_pc_set = true;
+  }
+
+  bool get_is_pc_set() const { return is_pc_set; }
```

Comment:
I accept the const-ness bit, but I don't see why disabling the default constructor is bad when default-constructing the class is actually a bad idea (as this is what the problem behind this PR is about).